paimon-core/src/main/java/org/apache/paimon/table/Table.java

```diff
@@ -74,6 +74,9 @@ default String fullName() {
     @Experimental
     Optional<Statistics> statistics();
 
+    @Experimental
+    Optional<Statistics> statistics(long snapshotId);
```

Comment:
Can we avoid introducing this method?

Maybe we can `Table.copy('scan.snapshot-id', 'xxx').statistics` to get stats for snapshot.

Reply:
Nice reminder，from this way can implement time travel for statistic. Had change it.